oap-server/server-core/src/test/java/org/apache/skywalking/oap/server/core/query/AggregationQueryServiceTest.java

```diff
@@ -0,0 +1,201 @@
+    private static final String INS_NAME = "my-ins";
+    private static final int SERVICE_ID = 12;
+    private static final String C_NAME = "my-c-name";
+
+    private static final int N = 10;
+
+    private static final Step STEP = Step.MINUTE;
+
+    private static final Order ORDER = Order.ASC;
+
+    @Before
+    public void setUp() throws Exception {
+
+        // mock ValueColumnIds
+        ValueColumnIds.INSTANCE.putIfAbsent(INS_NAME, C_NAME, Function.None);
+
+        // mock moduleServiceHolder
+        when(moduleServiceHolder.getService(IAggregationQueryDAO.class)).thenReturn(iAggregationQueryDAO);
+
+
+    }
+
+    @Test
+    public void getServiceTopN() throws Exception {
+
+        List<TopNEntity> mockEntities = mockTopNForDAO();
+        List<TopNEntity> topNEntities = aggregationQueryService.getServiceTopN(INS_NAME, N, STEP, START_TB, END_TB, ORDER);
+
+        assertTopNEntities(mockEntities, topNEntities, SERVICE_INVENTORY_NAME);
+    }
+
+    @Test
+    public void getServiceTopNWithEmpty() throws Exception {
+        mockTopNForDAO(Collections.emptyList());
+        List<TopNEntity> topNEntities = aggregationQueryService.getServiceTopN(INS_NAME, N, STEP, START_TB, END_TB, ORDER);
+        assertEquals(0, topNEntities.size());
+    }
+
+    @Test
+    public void getAllServiceInstanceTopN() throws Exception {
+
+        List<TopNEntity> mockEntities = mockTopNForDAO();
+        List<TopNEntity> topNEntities = aggregationQueryService.getAllServiceInstanceTopN(INS_NAME, N, STEP, START_TB, END_TB, ORDER);
+        assertTopNEntities(mockEntities, topNEntities, SERVICE_INSTANCE_INVENTORY_NAME);
+    }
+
+    @Test
+    public void getServiceInstanceTopN() throws Exception {
+        List<TopNEntity> mockEntities = mockTopNForDAO();
+        List<TopNEntity> topNEntities = aggregationQueryService.getServiceInstanceTopN(SERVICE_ID, INS_NAME, N, STEP, START_TB, END_TB, ORDER);
+        assertTopNEntities(mockEntities, topNEntities, SERVICE_INSTANCE_INVENTORY_NAME);
+    }
+
+    @Test
+    public void getAllEndpointTopN() throws Exception {
+        List<TopNEntity> mockEntities = mockTopNForDAO();
+        List<TopNEntity> topNEntities = aggregationQueryService.getAllEndpointTopN(INS_NAME, N, STEP, START_TB, END_TB, ORDER);
+        assertTopNEntities(mockEntities, topNEntities, ENDPOINT_INVENTORY_NAME);
+    }
+
+    @Test
+    public void getEndpointTopN() throws Exception {
+        List<TopNEntity> mockEntities = mockTopNForDAO();
+        List<TopNEntity> topNEntities = aggregationQueryService.getEndpointTopN(SERVICE_ID, INS_NAME, N, STEP, START_TB, END_TB, ORDER);
+        assertTopNEntities(mockEntities, topNEntities, ENDPOINT_INVENTORY_NAME);
+    }
+
+
+    private List<TopNEntity> mockEntities() {
+        List<TopNEntity> entities = new ArrayList<>(N);
+
+        for (int i = 0; i < N; i++) {
+            TopNEntity en = new TopNEntity();
+            en.setId(String.valueOf(i));
+            en.setName("name-" + i);
```

Comment:
The name should come from mock inventory, right?  You put a name here without any comment, make reader very hard to understand. Suggest you put a comment,
> Mock the entity name. This name should be replaced by mock inventory.